doc/customized+model.md

```diff
@@ -0,0 +1,130 @@
+# Customized Model
+
+SQLFlow supports training and predicting using customized models. This documentation explains the design choice and provides a concrete example of adding a new model.
+
+## Keras over Estimator
+
+We choose Keras over Estimator for the following reasons:
+
+1. TensorFlow 2.x will closely integrate with Keras. [ref](https://www.youtube.com/watch?v=k5c-vg4rjBw)
+
+2. Keras provides more documentation in writing customized models than estimators. For customized estimators, I've only found two examples:
+
+   1. `DNNClassifier` that uses `core_layers.Dense`. [ref](<https://github.com/tensorflow/estimator/blob/master/tensorflow_estimator/python/estimator/canned/dnn.py#L200-L226>)
+   2. `DNNClassifier` that uses `tf.layers.dense`. [ref](https://github.com/tensorflow/models/blob/master/samples/core/get_started/custom_estimator.py#L29)
+
+   None of these is suitable for long term development.
+
+3. There are plenty off-the-shelf models written in Keras. To name a few: [ResNet](https://github.com/raghakot/keras-resnet), [Transformer](https://github.com/Lsdefine/attention-is-all-you-need-keras), [Mask  R-CNN](https://github.com/matterport/Mask_RCNN).
+
+## Keras Model API
+
+Keras provides three major ways to define models:
+
+- Subclassing `tf.keras.Model`
+
+  ```python
+  class DNNClassifier(tf.keras.Model):
+      def __init__(self, feature_columns, hidden_units, n_classes):
+          super(DNNClassifier, self).__init__()
+          self.feature_layer = tf.keras.layers.DenseFeatures(feature_columns)
+          self.hidden_layers = []
+          for hidden_unit in hidden_units:
+              self.hidden_layers.append(tf.keras.layers.Dense(hidden_unit))
+          self.prediction_layer = tf.keras.layers.Dense(n_classes, activation='softmax')
+  
+      def call(self, inputs):
+          x = self.feature_layer(inputs)
+          for hidden_layer in self.hidden_layers:
+              x = hidden_layer(x)
+          return self.prediction_layer(x)
+  
+  model = DNNClassifier(feature_columns, hidden_units, n_classes)
+  ```
+
+  Please be aware that models subclassing from `tf.keras.Model` only supports `save_weights` and `load_weights`. [ref1](https://stackoverflow.com/questions/51806852/cant-save-custom-subclassed-model), [ref2](https://stackoverflow.com/questions/52826134/keras-model-subclassing-examples).
```

Comment:
I was confused by the point here. Obviously, subclasses support many more than `save_weights` and `load_weights` -- we can see `call` and `__init__`.

After reading the links you provided, I got that you didn't complete your sentence, which seems should have been:

> Please be aware that Keras Model subclasses allow us to save and load model parameters but not model topologies.